File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    
    public int[] TwoSum(int[] nums, int target) {
        int currentIndex = 0;
        int comparedIndex;

        //check if the pointer has reached the end of the list
        while (currentIndex < nums.Length)
        {
            comparedIndex = nums.Length - 1;
            // only compare later values (greater index)
            while (currentIndex < comparedIndex)
            {
            
                if (nums[currentIndex] + nums[comparedIndex] == target)
                {
                    return new int[] { currentIndex, comparedIndex };
                }
                comparedIndex -= 1;
            }
            currentIndex += 1;
        }
        return new int[] { 0, 0 };
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {}
        for ind, num in enumerate(nums):
            a_number_needed = target - num
            if a_number_needed in hash_map:
                return [hash_map[a_number_needed], ind]
            else:
                hash_map[num] = ind
                


        